packages/grafana-ui/src/components/VizLegend/VizLegendListItem.tsx

```diff
@@ -100,6 +101,9 @@ const getStyles = (theme: GrafanaTheme2) => ({
     font-size: inherit;
     padding: 0;
     user-select: text;
+    max-width: 600px;
```

Comment:
Where did we get 600 value from? I think this works well for when legend is placed underneath but not as great when placed to the right

I think this is a definite improvement over previous behavior so it's worth merging in before we have a perfect solution - we may want to consider a few things in the future like:
- Allowing user to set max-column width / width of legend item
- Responsive max column widths based on content / layout
- etc.

Reply:
@nmarrs I agree, that's why I think would be great to have a UX (@lukasztyrala) session to come up with an overall template, since this is impacting other panels too. We already have a `width` option, but I don't think it works as expected (or at least the way I would expect). ^Same thing for tooltips.

The old graph is using a max-width of 650px. Also the legend has a max-width of 60%, which I think looks a bit weird for right placement.. so definitely lots of things to improve here!😅